src/Microsoft.ML.Tokenizers/Tokenizer.cs

```diff
@@ -136,6 +137,86 @@ public int CountTokens(string text, bool considerSpecialTokens = true)
             return idsCount;
         }
 
+        /// <summary>
+        /// Find the maximum encoding capacity from beginning within the input text without surpassing the token limit.
+        /// </summary>
+        /// <param name="text">The text to encode.</param>
+        /// <param name="maxTokenCount">The maximum token count to limit the encoding capacity.</param>
+        /// <param name="considerSpecialTokens">Indicate if want to consider the special tokens during the encoding.</param>
+        /// <returns>
+        /// - The entire normalized text.
+        /// - The length of text from the beginning of the normalized which is limited by the maximum token count
+        /// - The token count can be generated using the provided length which should be smaller than the maximum token count.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">The input text is null.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">The maximum token count must be greater than 0.</exception>
+        /// <remarks>
+        /// If the tokenizer has a normalizer, the returned text will be the normalized text. Otherwise the returned text will be the input text.
+        /// If the provided <paramref name="maxTokenCount"/> is greater than the token count of the input text, the returned length will be the length of the input text.
+        /// If the provided <paramref name="maxTokenCount"/> is smaller enough to hold smallest number of grouped Ids, the returned length will be 0 and returned TokenCount will be 0.
+        /// </remarks>
+        public (string Text, int Length, int TokenCount) TrimSuffixWithinTokenLimit(string text, int maxTokenCount, bool considerSpecialTokens = true)
```

Comment:
Do we need the "WithinTokenLimit" part of the name? I don't have a strong opinion; just not sure if it adds anything meaningful.

Reply:
My issue with naming is the word `Trim` indicates we return the trimmed text while we return the whole text with either Length or Offset. If you have a better name suggestion that will be very helpful.

Reply:
That's a fair concern, but I don't think "WithinTokenLimit" addresses that concern.

Reply:
I am not suggesting to keep `WithinTokenLimit`. I am just looking for a better method name :-)

Reply:
Here is what the AI suggested :-) `FindEncodableLengthWithinLimit`

Reply:
Thinking more, I think we can go with the method names, `FindPrefixEncodableLength` and `FindSuffixEncodableOffset`. What do you think about that?

Reply:
We already decide about `IndexOfTokenCount` and `LastIndexOfTokenCount` with changing the method's signatures.